### ISSUE TITLE
RadioGroup - Rounded

### ISSUE THREAD

**Issue by asangma** [Esri/calcite-components#743]
I propose a `rounded` property on RadioGroup.

We are going to see an increasing number of cases where a decision point in a workflow will need a radio group, but that moment may not benefit from the horizontal separation that RadioGroup currently delivers.

In this case, the options below the pills change depending on which item is selected.

cc @mitc7862 @bstifle @macandcheese @paulcpederson

**Comment by bstifle** [comment]
i wouldn't have them separated like that. they look just like chips tags/filtering

**Comment by macandcheese** [comment]
IMO these look like chips, where more than one can be selected - not a single-choice radio group. They also look like the `+ field comparison` buttons (?) above.

> but that moment may not benefit from the horizontal separation that RadioGroup currently delivers.

What does this mean exactly?

**Comment by bstifle** [comment]
we already have this square design in the works, we could potentially add a round version i suppose.... though it seems a little too "apple"

![image](https://user-images.githubusercontent.com/14083530/87723846-3ef8a000-c76f-11ea-8180-f746c99687d0.png)

**Comment by asangma** [comment]
One of the issues with the radio group in its current form and in this use case is that it behaves as a horizontal separator.

Also, I think it's OK that they look like the other buttons.
Recall that in their default design they also look like out regular buttons.

**Comment by bstifle** [comment]
@asangma 

i just think it might be a combo of feature creep on radio groups, as well as a style creep on chips.

can you use normal radio buttons?

![image](https://user-images.githubusercontent.com/14083530/87724592-93e8e600-c770-11ea-8e49-8c4d637fd967.png)

**Comment by macandcheese** [comment]
> Recall that in their default design they also look like out regular buttons.

Yeah, there's an issue to update the radio group to Bryan's example above (sans rounded version) to alleviate that.
https://github.com/Esri/calcite-components/issues/639

The above design can pretty much be achieved with chips, but we don't really have the concept of "selectable" chips (ie a developer would need to handle the interaction amongst the chips):
<img width="352" alt="Screen Shot 2020-07-16 at 2 27 01 PM" src="https://user-images.githubusercontent.com/4733155/87724531-7c116200-c770-11ea-902b-d3b6164ce0b3.png">

**Comment by asangma** [comment]
Thanks @macandcheese.

@bstifle We totes mcgoats could use regular radio buttons.
The intention here is to convey that those options have significant effects on the results of this workflow.

Just looking for a way to have the weight of the radio-group while not adding another horizontal divider...or one with a weird width.

![image](https://user-images.githubusercontent.com/12503298/87725190-a9124480-c771-11ea-9dbd-4ba03f8a8d7a.png)

Or full-width:
![image](https://user-images.githubusercontent.com/12503298/87725390-145c1680-c772-11ea-909e-426e53d04f27.png)

**Comment by asangma** [comment]
As a note, for "First feature" there would be no options below it.

**Comment by bstifle** [comment]
cool, so the examples you just made up there, you could use the Small prop, and the height is around 32px with 14px type. little more petite

**Comment by asangma** [comment]
![image](https://user-images.githubusercontent.com/12503298/87725687-9d734d80-c772-11ea-94fc-674b7c3c3e6a.png)

**Comment by asangma** [comment]
@bstifle Wouldn't I still get a radio-group that is either a weird width or full-width and acting as another horizontal separator?

**Comment by macandcheese** [comment]
I think the concern is that those look just like chips, which currently don't have any "selected" state / toggle-ability. You could just use chips and change color / appearance type on selection, but even then I feel like a "chip selection" workflow insinuates a multi-select, not a single-select.

I see what you mean about "horizontal separator", but I don't think the "fit to size of options" one causes any more of an issue than the side-by-side inputs acting in a similar fashion. I think generally there are a lot of UI shapes, sizes, and column widths of elements going on in the workflow, so to that end I agree that the radio group wouldn't do much to help there.

**Comment by bstifle** [comment]
i still recommend normal radio buttons if the concern lies with too many lines/separators

**Comment by bstifle** [comment]
there are plenty of input fields in there that add to the lines and such, so i'm not really seeing the argument for not using this fella

![image](https://user-images.githubusercontent.com/14083530/87726573-1c1cba80-c774-11ea-9330-2380cc7002ba.png)

**Comment by bstifle** [comment]
it could, but with all these options proposed i still recco this new radio group design 
![image](https://user-images.githubusercontent.com/14083530/87726699-630ab000-c774-11ea-88e5-dc96402f0ae0.png)


sorry for the redundancy haha

**Comment by asangma** [comment]
> i'm not really seeing the argument for not using this fella

I think if when it acts as a kinna header for a section, it works well.
In this case, it is inside a section and messes with the visual hierarchy.

**Comment by bstifle** [comment]
but that's literally what it was designed for 😭  changing/revealing content beneath it

**Comment by bstifle** [comment]
i guess you could use Tabs as well

**Comment by asangma** [comment]
Indeed. I get that.

In the case of "First feature" it'd look something like this.
![image](https://user-images.githubusercontent.com/12503298/87727492-e8db2b00-c775-11ea-8179-4f1d31406bcc.png)

**Comment by bstifle** [comment]
![image](https://user-images.githubusercontent.com/14083530/87729112-72d8c300-c779-11ea-9f54-44c44300c6de.png)

so does "Feature Inclusion" change what populates in "Result Layer Fields"? if so, do you need the line?

**Comment by asangma** [comment]
Yeah...those are two separate sections.

**Comment by asangma** [comment]
That's kinna what I'm talking about in regards to the visual hierarchy and radio-group adding a horizontal divider.

**Comment by asangma** [comment]
So if there's resistance to the rounded approach, I think we need to rework the existing design some.
In the interface, it still looks pretty similar to CalciteButton.

I know there is a 👎  on the [rounded look](https://github.com/Esri/calcite-components/issues/743#issuecomment-659679051), but in an interface, it works.
![image](https://user-images.githubusercontent.com/12503298/90149599-83a93400-dd39-11ea-95ac-6218edc742ed.png)

**Comment by macandcheese** [comment]
Unrelated.. Could we figure out a way to make a Calcite-stepper work for this workflow (no icon, vertical, and not setting "complete" on any steps would be really close to this)...

**Comment by asangma** [comment]
@macandcheese #862

**Comment by asangma** [comment]
Following up on this...the radio group falls apart in panels.
We could use a pattern that perhaps allows the items to wrap.

![Screen Shot 2020-10-01 at 4 14 43 PM](https://user-images.githubusercontent.com/12503298/94872362-4f7cf600-0401-11eb-92ed-c9daac5a36c3.png)

We ended up with something more like
![image](https://user-images.githubusercontent.com/12503298/94872589-de8a0e00-0401-11eb-9758-5f09f5261f7b.png)

**Comment by asangma** [comment]
Wraps like this...not amazing, but works.
![image](https://user-images.githubusercontent.com/12503298/94872855-93242f80-0402-11eb-82c0-75fe5081dfda.png)

**Comment by bstifle** [comment]
Can something like this work? also suggest using Small instead of Medium

![image](https://user-images.githubusercontent.com/14083530/94945548-73434900-04a0-11eb-863b-f16c105a5859.png)

**Comment by macandcheese** [comment]
There's also vertical layout mode, a bit extra line height... But, yes generally panels seem very narrow to accommodate more than two things in horizontal lockups, so scale "s" would certainly help as well.

**Comment by bstifle** [comment]
Right yeah, vert:

![Uploading image.png…]()

**Comment by bstifle** [comment]
Right yeah, vert:

![image](https://user-images.githubusercontent.com/14083530/94946325-90c4e280-04a1-11eb-8b61-2a58ac662330.png)

**Comment by macandcheese** [comment]
A slightly wider panel option might help a lot of these UI breathe a bit more too? I know there was talk of a "really wide" version but maybe an "m" between this and that?

**Comment by asangma** [comment]
re: panel width
Panel widths will be highly dynamic, and our designs and components will need to handle narrow panels. Also, consider what happens if there's more than three options...or when we're localizing to German or Latvian.

re: scale
`scale="s"` is the first thing we tried, but it's still brittle. We will still run into the width problem when there are more options, when an option has a slightly longer string, or when localizing. 

re: wrapping
@bstifle 
> Can something like this work?

![image](https://user-images.githubusercontent.com/14083530/94945548-73434900-04a0-11eb-863b-f16c105a5859.png)

We tried that. And it's a quick technical fix. Would we want to do that? Like would we do that in a mobile interface?

re: Vertical
Seems like we're losing the intent of the design in this case. It also starts to look like a stack of buttons too similar to a Done & Cancel pattern. Not sure how well it works in the panel.

![image](https://user-images.githubusercontent.com/12503298/94973451-9c73d180-04c0-11eb-85d8-ccc6fd5a3337.png)


I think we need more UI options with radio-group.
